rpms/awips2.edex/Installer.edex/programs/updateNDM.pl

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/perl
+
+#This script gets installed with EDEX and a cronjob is setup to run this script daily. This script initially sets up a local git repo and is configured to just checkout the National Dataset files from Unidata's public git repo and then copies the files to /awips2/edex/data/ndm to keep the databases updated with new metar/station information.
+
+$ndmGitPath = "rpms/awips2.edex/Installer.edex/ndm/";
+
+#get the current branch/version of awips2
+$version = `grep AWIPSII_VERSION /awips2/repo/awips2/build/buildEnvironment.sh`;
+chomp $version;
+@split = split(/"/,$version);
+
+$branch = "unidata_$split[1]";
+
+#Check if ndm location has been created
+$ndmDir="/awips2/dev/ndm";
+if(!-d $ndmDir)
+{
+  `mkdir $ndmDir`;
+}
+
+#Check if ndm git files have been checked out yet
+if(!-e "$ndmDir/.git")
+{
+  `cd $ndmDir ; 
+   git init ; 
+   git remote add -f origin https://github.com/Unidata/awips2.git ; 
+   git config core.sparseCheckout true ;
+   echo \"$ndmGitPath\" >> .git/info/sparse-checkout ;
+   git checkout -t origin/$branch
+  `;
+}
+
+print "\n\n\n";
+
+#Pull latest files
+`cd $ndmDir ; 
+ git pull origin $branch
```

Comment:
So... `git pull` sometimes gets in weird states for me... Like if, for some reason, the user manually changes some of these files.  Then even with a `git pull` the local repo can still be out of sync or ahead of the main repo... Maybe it's fine and a risk we're willing to take, but if we really want to make sure we have exactly what's in our public repo then this line would be safer:
`git reset --hard origin/$branch`

Reply:
Okay, sure - that's a good suggestion. So run that before the git pull? 

Also want to add a `git fetch` - should that go before or after the `git reset`?

Reply:
so with `git reset --hard origin/$branch` you don't need any `git pull`.

I'm not entirely sure a `git fetch` would be needed either... but to be safe, you'd run the `fetch` before the `reset`